solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/EuristicBoostedModel.java

```diff
@@ -0,0 +1,150 @@
+        type = (String) featureBoostParam.get(BOOST_TYPE);
+      }
+    }
+
+    private void checkParams(Map<String, Object> euristicBoostParams) {
+      checkParam(euristicBoostParams, FEATURE_NAME);
+      checkParam(euristicBoostParams, WEIGHT);
+    }
+
+    private void checkParam(Map<String, Object> featureBoostParams, String paramName) {
+      String modelName=EuristicBoostedModel.this.getName();
+      if (!featureBoostParams.containsKey(paramName)) {
+        throw new ModelException("Model " + modelName + " required param " + paramName + " not defined");
+      }
+    }
+
+    private int getFeatureIndex( String featureName) {
+      String modelName=EuristicBoostedModel.this.getName();
+      final List<Feature> features = getFeatures();
+      Optional<Feature> featureToBoostOptional = features.stream()
+          .filter(f -> f.getName().equals(featureName))
+          .findFirst();
+
+      if (featureToBoostOptional.isPresent()) {
+        Feature featureToBoost = featureToBoostOptional.get();
+        return featureToBoost.getId();
+      } else {
+        throw new ModelException("Model " + modelName + " doesn't contain any feature with the name=[" + feature + "}");
+      }
+    }
+  }
+
+  public EuristicBoostedModel(String name, List<Feature> features,
+                              String featureStoreName, List<Feature> allFeatures,
+                              NamedParams params) throws ModelException {
+    super(name, features, featureStoreName, allFeatures, params);
+    initEuristicBoost();
+  }
+
+  protected void initEuristicBoost() {
+    Map<String, Object> params = (Map<String, Object>) EuristicBoostedModel.this.getParams().get(EURISTIC_BOOST_PARAM);
+    if (params != null) {
+      this.euristicFeatureBoost = new EuristicFeatureBoost(params);
+    }
+  }
+
+  @Override
+  public float score(float[] modelFeatureValuesNormalized) {
+    internalModelScore = LTRModel.score(modelFeatureValuesNormalized);
+    float finalScore = internalModelScore;
+    if (euristicFeatureBoost != null) {
+      boostFeatureValue = modelFeatureValuesNormalized[euristicFeatureBoost.featureIndex];
+      weightedBoostFeatureValue = euristicFeatureBoost.weight * boostFeatureValue;
+
+      if (MULTIPLICATIVE_BOOST.equals(euristicFeatureBoost.type)) {
+        if (internalModelScore < 0) {
+          weightedBoostFeatureValue = 1 / weightedBoostFeatureValue;
+        }
+        finalScore *= weightedBoostFeatureValue;
+      } else if (ADDITIVE_BOOST.equals(euristicFeatureBoost.type)) {
+        finalScore += weightedBoostFeatureValue;
+      }
+
+    }
+    return finalScore;
+  }
+
+  @Override
+  public Explanation explain(LeafReaderContext context, int doc, float finalScore, List<Explanation> featureExplanations) {
+    if (euristicFeatureBoost == null) {
+      return LTRModel.explain(context, doc, internalModelScore, featureExplanations);
+    }
+
+    String boostTypeOperator = "prod";
```

Comment:
I would be consistent with the boost type operators we have instead of adding new text representations for them.  Rename ADDITIVE_BOOST and MULTIPLICATIVE_BOOST  to  "sum" and "product", and then you can just use euristicFeatureBoost directly instead of having these 3 lines.  Solr function queries also use "sum" and "product", so it's somewhat consistent with that.